cmd/node/factory/structs.go

```diff
@@ -981,93 +991,176 @@ func createShardDataStoreFromConfig(
 		}
 	}()
 
-	txUnit, err = storageUnit.NewStorageUnitFromConf(
-		getCacherFromConfig(config.TxStorage.Cache),
-		getDBFromConfig(config.TxStorage.DB, uniqueID),
-		getBloomFromConfig(config.TxStorage.Bloom))
+	fullArchiveMode := config.StoragePruning.FullArchive
+	numOfEpochsToKeep := uint32(config.StoragePruning.NumOfEpochsToKeep)
+	numOfActivePersisters := uint32(config.StoragePruning.NumOfActivePersisters)
+
+	txUnitStorerArgs := &pruning.PruningStorerArgs{
```

Comment:
you could use only one PruningStorerArgs, and change only those things which are different.

Reply:
didn't change this. half of the params are unique so I'd have to change them always. Also I think it is safer that we have a new arguments instance for all the storers